inc/ruleticket.class.php

```diff
@@ -238,6 +238,34 @@ function executeActions($output, $params, array $input = []) {
                      }
                   }
 
+                  if ($action->fields["field"] == 'task_template') {
+                     $template = new TaskTemplate();
+                     if ($template->getFromDB($action->fields["value"])) {
+                        $ticket_id = $output['id'] ?? 0;
+                        if (!$ticket_id) {
+                           // Target ticket is not yet created, data need to be
+                           // moved to special input '_tasktemplates_id' so it
+                           // can be handled after the ticket creation
+                           $output["_tasktemplates_id"][] = $template->getId();
+                        } else {
```

Comment:
In the same spirit as @cedric-anne remark, we should have only `$output["_tasktemplates_id"][] = $template->getId();` also for update.

The code here is the same as `CommonItilObject::post_addItem`.

So, i think the solution is to add a new method in `CommonItilObject` to manage creation of task from `_tasktemplates_id` input and call it in both functions `post_xxxItem`

You will be also able to do real **unit** test (currently you do a functional one based on rules)

Reply:
> In the same spirit as @cedric-anne remark, we should have only `$output["_tasktemplates_id"][] = $template->getId();` also for update.
> 
> The code here is the same as `CommonItilObject::post_addItem`.
> 
> So, i think the solution is to add a new method in `CommonItilObject` to manage creation of task from `_tasktemplates_id` input and call it in both functions `post_xxxItem`
> 
> You will be also able to do real **unit** test (currently you do a functional one based on rules)

Done for the code change, I agree.

For the unit test, at this moment it does not seem possible for me to have a correct one as this method use multiple "repository" objects and we lack dependency injection to mock them properly.